### DISCUSSION TITLE
Ambiguous Call

### DISCUSSION THREAD

**Discussion by rodrigo-machado-atg** [danielpalme/ReportGenerator#438] [Q&A]
Hi there,

I'm getting an ambiguous call error when using ReportGenerator v. 4.8.12.

My call is as follow:

...
 var reportGeneratorSettings = new ReportGeneratorSettings()
            {
                HistoryDirectory = new DirectoryPath("./TestReports/ReportsHistory")
            };

            ReportGenerator( "./TestReports/coverage.xml" , "./TestReports/ReportGeneratorOutput", reportGeneratorSettings);
			var reportFilePath = "./TestReports/ReportGeneratorOutput/index.htm";
...

And the error message is:

error CS0121: The call is ambiguous between the following methods or properties: 'ReportGenerator(GlobPattern, DirectoryPath, ReportGeneratorSettings)' and 'ReportGenerator(FilePath, DirectoryPath, ReportGeneratorSettings)'

There are two methods with the same or similar signature and ReportGenerator doesn't know how to discerne between them.

Am i doing something wrong?

Thanks in advance

**Comment by danielpalme** [comment]
Your code should probably look like this:
```csharp
new Palmmedia.ReportGenerator.Core.Generator().GenerateReport(
    new ReportConfiguration(
        new[] { "./TestReports/coverage.xml" },
        "./TestReports/ReportGeneratorOutput",
        new string[0],
        "./TestReports/ReportsHistory",
        new string[0],
        new string[0],
        new string[0],
        new string[0],
        new string[0],
        null,
        null));
```
